library/choose_disk.py

```diff
@@ -0,0 +1,436 @@
+
+        if ceph_disk is True:
+            selected_devices[physical_disk]['ceph'] = 1
+            info(' Adding   %10s : Ceph disk detected' % physical_disk)
+        else:
+            info(' Adding   %10s : %s' % (physical_disk, selected_devices[physical_disk]['bdev']))
+
+    return selected_devices
+
+
+def get_block_devices_persistent_name(physical_disks):
+    ''' Replace the short name (sda) by the persistent naming 'by-id' '''
+    directory = "/dev/disk/by-id/"
+
+    info('Finding persistent disks name')
+    # If the directory doesn't exist, reports the list as-is
+    if not os.path.isdir(directory):
+        info(' Cannot open %s' % directory)
+        return physical_disks
+
+    final_list = {}
+    matching_list = {}
+    for f in os.listdir(directory):
+        device_name = os.readlink(directory + f).split("/")[-1]
+        if device_name in physical_disks:
+            if device_name not in matching_list:
+                matching_list[device_name] = [f]
+            else:
+                matching_list[device_name].append(f)
+
+    for physical_disk in sorted(physical_disks):
+        if physical_disk in matching_list:
+            current_index = sorted(matching_list[physical_disk])[0]
+            final_list[current_index] = physical_disks[physical_disk]
+            final_list[current_index]["bdev"] = "%s%s" % (directory, current_index)
+            info(' Renaming %10s to %50s' % (physical_disk, current_index))
+        else:
+            current_index = physical_disk
+            final_list[current_index] = physical_disks[physical_disk]
+
+    return final_list
+
+
+def fake_device(device_list):
+    '''
+    In case of legacy block device names, let's create an internal faked
+    entry with a 'bdev' entry filled with the actual path. This will be used to
+    make a match later on.
+    '''
+    devices = {}
+    count = 0
+    for device in device_list.split():
+        devices["legacy_%d" % count] = {"bdev": os.path.dirname(device)+"/"+os.path.basename(device)}
+        count = count + 1
+
+    return devices
+
+
+def show_resulting_devices(matched_devices, physical_disks):
+    bdev_matched = []
+    bdev_unmatched = []
+    info("Matched devices   : %3d" % len(matched_devices))
+    for matched_device in sorted(matched_devices.keys()):
+        extra_string = ""
+        if "ceph" in matched_devices[matched_device]:
+            extra_string = " (ceph)"
+        info(" %s : %s%s" % (matched_device, matched_devices[matched_device]["bdev"], extra_string))
+        bdev_matched.append(matched_devices[matched_device]["bdev"])
+
+    for physical_disk in sorted(physical_disks):
+        if physical_disks[physical_disk]["bdev"] not in bdev_matched:
+            bdev_unmatched.append(physical_disks[physical_disk]["bdev"])
+
+    info("Unmatched devices : %3d" % len(bdev_unmatched))
+    for bdev in sorted(bdev_unmatched):
+        info(" %s" % bdev)
+
+
+def info(message):
+    global logger
```

Comment:
@ErwanAliasr1 I don't think you'd need to pass the logger to every call though. Anywhere you're calling ``info("some message")`` this should be used instead: ``logger.info("some message")``. Doing so would remove the need for all of these logging functions.